internal/consensus/state.go

```diff
@@ -314,7 +314,12 @@ func (cs *State) LoadCommit(height int64) *types.Commit {
 	defer cs.mtx.RUnlock()
 
 	if height == cs.blockStore.Height() {
-		return cs.blockStore.LoadSeenCommit(height)
+		commit := cs.blockStore.LoadSeenCommit()
+		// NOTE: we can't yet ensure atomicity of operations in asserting
+		// whether this is the latest height and retrieving the seen commit
```

Comment:
```suggestion
		// NOTE: We can't yet ensure atomicity of operations in asserting
		// whether this is the latest height and retrieving the seen commit.
```

Reply:
Not sure I totally understand this comment.  Maybe:

```
NOTE: Retrieving the height of the most recent block and retrieving the most recent commit does not currently occur as an atomic operation. We check the height and commit here in case a more recent commit has arrived since retrieving the latest height.
```